homeassistant/components/homekit/accessories.py

```diff
@@ -0,0 +1,55 @@
+"""Extend the basic Accessory and Bridge functions."""
+from pyhap.accessory import Accessory, Bridge, Category
+
+from .const import (
+    SERVICES_ACCESSORY_INFO, MANUFACTURER,
+    CHAR_MODEL, CHAR_MANUFACTURER, CHAR_SERIAL_NUMBER)
+
+
+class HomeAccessory(Accessory):
+    """Class to extend the Accessory class."""
+
+    ALL_CATEGORIES = Category
+
+    def __init__(self, display_name):
+        """Initialize a Accessory object."""
+        super().__init__(display_name)
+
+    def set_category(self, category):
+        """Set the category of the accessory."""
+        self.category = category
+
+    def set_services(self, services):
```

Comment:
Note that usually the services are defined by overriding `_set_services` (and calling the default implementation, which adds the AccessoryInfo Service). This is because the services must be all added before the accessory is paired with an iOS client, otherwise it will see only part of the services before it refetches the whole Accessory info again.

Also note that the AccessoryInformation service **must** have an IID of 1 (required by HAP) - currently this can only be ensured if the default `_set_services` is called before other services are added (this will be fixed soon).